src/routes/json-api/SettingsRoute.ts

```diff
@@ -11,39 +11,39 @@ import { Validator } from '../../validators/Validator';
 
 const router = Router();
 
-router.get('/hiddenSettings', (_, res : Response) => {
-    res.json(
-        {
-	 'HIDE_DONATE' : process.env.HIDE_DONATE || false,
-            'VERSION' : version
-        }
-    )
+router.get('/hiddenSettings', (_, res: Response) => {
+    res.json({
+        HIDE_DONATE: Boolean(process.env.HIDE_DONATE) || false,
+        VERSION: version,
+    });
 });
 
-router.get('/', async (_, res : Response) => {
-    const configMap : ConfigMap = await configService.getAllConfig();
+router.get('/', async (_, res: Response) => {
+    const configMap: ConfigMap = await configService.getAllConfig();
     res.json(configMap);
 });
 
-router.put('/', async (req : Request, res : Response) => {
-    const validator : Validator = new ConfigFormValidator();
-    const validationResult : {[key:string] : string} = await validator.validate(req.body);
-    if (Object.keys(validationResult).length > 0){
-        const apiResponse : ApiResponse = {
-            error : ApiError.INVALID_INPUT,
-            invalid_fields : validationResult
-        }
+router.put('/', async (req: Request, res: Response) => {
+    const validator: Validator = new ConfigFormValidator();
+    const validationResult: { [key: string]: string } = await validator.validate(req.body);
+    if (Object.keys(validationResult).length > 0) {
+        const apiResponse: ApiResponse = {
+            error: ApiError.INVALID_INPUT,
+            invalid_fields: validationResult,
+        };
         res.status(400).json(apiResponse);
         return;
     }
-    for (const key of Object.keys(req.body)){
-        if (key != 'AUTH_PASSWORD'){
-            await configService.setParameter(key as IplayarrParameter, req.body[key]);
-        } else {
-            const existingPassword = await configService.getParameter(IplayarrParameter.AUTH_PASSWORD);
-            if (existingPassword != req.body[key]){
```

Comment:
This was the bug, the existing password is a hashed value, so it'll never match unless they decide to set their previous hash value as their new password 🙃